Setup privy sdk

Nathaniyell opened the pull request:
<!-- This is an auto-generated comment: release notes by coderabbit.ai -->

## Summary by CodeRabbit

- **New Features**
	- Introduced the `PrivyProvider` component for enhanced authentication options, allowing users to log in via email or wallet.
	- Configured UI customization options for a better user experience.

- **Chores**
	- Added `@privy-io/react-auth` as a new dependency for authentication management.
	- Updated `.gitignore` to exclude `package-lock.json` from version control.

<!-- end of auto-generated comment: release notes by coderabbit.ai -->

Nathaniyell committed:
privy provider setup complete

sweetmantech commented on providers/PrivyProvider.tsx:
```diff
@@ -0,0 +1,29 @@
+const PrivyProvider = ({ children }: { children: ReactNode }) => {
+  return (
+    <Provider
+    appId={process.env.NEXT_PUBLIC_PRIVY_APP_ID || (() => {
```
too much logic inline.
- please declare a new variable `privyAppId` where you handle the `error` handling logic

Nathaniyell committed:
Refactor PrivyProvider to improve environment variable handling

- Extracted the retrieval of NEXT_PUBLIC_PRIVY_APP_ID into a separate variable for clarity.
- Added a check to throw an error if the environment variable is not set, ensuring better error handling.
